packages/expo-asset/src/resolveAssetSource.web.ts

```diff
@@ -7,19 +7,18 @@ declare let nativeExtensions: any;
 
 let _customSourceTransformer;
 let _serverURL: string | undefined | null;
-let _scriptURL: string | undefined | null;
 let _sourceCodeScriptURL: string | undefined | null;
 
 function getSourceCodeScriptURL(): string | undefined | null {
   if (_sourceCodeScriptURL) {
     return _sourceCodeScriptURL;
   }
 
-  let sourceCode = nativeExtensions && nativeExtensions.SourceCode;
+  let sourceCode = typeof nativeExtensions !== 'undefined' ? nativeExtensions.SourceCode : null;
```

Comment:
```suggestion
  let sourceCode = nativeExtensions?.SourceCode ?? null;
```

or did you want an `undefined` to leak through if `SourceCode` was `undefined`?